feat: 독서록 등록 기능

dani820 opened the pull request:
## 기능 설명
사용자가 작성한 독서록을 등록하는 기능입니다.

## 작업 내용 상세
- [x] 비즈니스 로직 작성
- [x] 단위 테스트 작성

## 참고 (옵션)

dani820 committed:
[#6] feat: 독서록 등록 기능 - 패키지 및 클래스 생성

- [ ] 비즈니스 로직 작성 (ing)
- [ ] 단위 테스트 작성 (ing)

f-lab-bradley commented on src/main/java/com/inmybook/domain/member/Member.java:
```diff
@@ -0,0 +1,15 @@
+import lombok.Getter;
+
+@Builder
+@Getter
```
도메인에는 필요한 부분에만 Getter 를 넣는걸로 하시죠

dani820 replied:
넵 수정하겠습니다

f-lab-bradley commented on src/main/java/com/inmybook/domain/member/Member.java:
```diff
@@ -0,0 +1,15 @@
+@Getter
+@AllArgsConstructor
+public class Member {
+	String memberId;
```
접근제어자가 없으면 캡슐화가 깨질 수 있어요

dani820 replied:
추가하겠습니다!

f-lab-bradley reviewed (commented):
포매터 적용도 필요 해 보이네요

dani820 committed:
[#6] feat: 독서록 등록 기능 - 패키지 및 클래스 생성

- [ ] 비즈니스 로직 작성 (ing)
- [ ] 단위 테스트 작성 (ing)

rebase

dani820 committed:
[#6] feat: 독서록 등록 기능

- [ ] 비즈니스 로직 작성 (ing)
- [ ] 단위 테스트 작성 (ing)

테스트 코드 추가

dani820 committed:
[#6] feat: 독서록 등록 기능

- [ ] 비즈니스 로직 작성 (ing)
- [ ] 단위 테스트 작성 (ing)

접근 제어자 추가

f-lab-bradley commented on src/main/java/com/inmybook/adapter/in/web/dto/ContentDto.java:
```diff
@@ -0,0 +1,16 @@
+package com.inmybook.adapter.in.web.dto;
+
+public record ContentDto(
+	String isbnNo,
```
Book 도 묶을 수 있을 것 같은뎅....

f-lab-bradley commented on src/main/java/com/inmybook/application/port/in/RegistPostCommand.java:
```diff
@@ -0,0 +1,25 @@
+@Builder
+public record RegistPostCommand(
+	String isbnNo,
+	String bookName,
```
명확한건 묶어도 될 것 같아요.
굳이 풀어 쓸 이유가 없는데..

f-lab-bradley commented on src/main/java/com/inmybook/application/port/in/RegistPostUsecase.java:
```diff
@@ -0,0 +1,5 @@
+package com.inmybook.application.port.in;
+
+public interface RegistPostUsecase {
```
interface 는 public 이 필요 없을껄요???

f-lab-bradley commented on src/main/java/com/inmybook/application/port/out/RegistPostPort.java:
```diff
@@ -0,0 +1,7 @@
+import com.inmybook.domain.post.Post;
+
+public interface RegistPostPort {
+	public int registPost(Post post);
```
인터페이스는 어떤 기능을 하고 어떤걸 파라미터로 받고 어떤걸 리턴하는지 주석으로 알려주는게 좋을 것 같아요.

dani820 replied:
설명 주석 추가하겠습니다!

f-lab-bradley commented on src/main/java/com/inmybook/application/service/RegistPostService.java:
```diff
@@ -0,0 +1,24 @@
+	@Override
+	public String registPost(RegistPostCommand registPostCommand) {
+		Post post = postFactory.createPost(registPostCommand);
+		int postNum = registPostPort.registPost(post);
```
등록하다의 동사형은 register 일거에요 아마...

dani820 replied:
수정하겠습니다~

f-lab-bradley commented on src/main/java/com/inmybook/application/service/RegistPostService.java:
```diff
@@ -0,0 +1,24 @@
+		Post post = postFactory.createPost(registPostCommand);
+		int postNum = registPostPort.registPost(post);
+
+		String path = "/post/" + postNum;
```
이런 url 규칙은 Post 도메인에 있는게 어울릴 것 같네요.

dani820 replied:
넵 도메인에서 생성하는 것으로 수정하겠습니다

f-lab-bradley commented on src/main/java/com/inmybook/domain/post/Post.java:
```diff
@@ -0,0 +1,29 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Post {
```
Post 가 너무 많은 역할을 하는 것 같아요.
Post 안에 다른 서브 도메인들을 설계 하는게 좋아 보이네요

f-lab-bradley reviewed (commented):
Mock 테스트를 잘 작성 해 주셨네요.

처음 해 본거라고는 생각 도 못할 정도로 잘 해 주셨어요!

dani820 committed:
[#6] feat: 독서록 등록 기능 클래스명 변경

- [ ] 비즈니스 로직 작성 (ing)
- [ ] 단위 테스트 작성 (ing)

클래스명 변경(regist -> register)

dani820 committed:
[#6] feat: 독서록 등록 기능 독서록 상세 조회 path 생성 로직 추가

Post 클래스 내 독서록 상세 조회를 위한 path 생성 함수 추가
RegisterPostPort 클래스 내 registerPost 리턴값 int -> String 변경
이에 따른 테스트 코드 수정

dani820 committed:
[#6] feat: 독서록 등록 기능 메서드 설명 주석 추가

기능 설명, 파라미터, 리턴값 총 3가지 항목으로 작성

dani820 committed:
Merge remote-tracking branch 'origin/develop' into feature/#6

f-lab-bradley reviewed (commented):
Force push 는 왜 하셨나요???

dani820 commented:
동일한 커밋 로그로 여러 번 커밋하게 되어서 합치는 편이 나을 것 같아 force push 하게 되었습니다

dani820 committed:
feat: 독서록 등록 기능 컨트롤러 작성 및 도메인 분리

dani820 committed:
feat: 독서록 등록 기능 도메인 로직 변경에 따른 테스트 코드 수정

썸네일 첨부한 경우와 첨부하지 않은 경우를 분리

dani820 committed:
feat: 인메모리 방식 repository 추가 및 테스트 코드 수정

f-lab-bradley commented on src/main/java/com/inmybook/adapter/in/web/PostController.java:
```diff
@@ -0,0 +1,56 @@
+		RegisterPostCommand registerPostCommand = postMapper.createRegisterPostCommand(registerPostInput,
+			multipartFile);
+		String path = "";
+		try {
```
이건 UseCase 에서 잡아주는게 좋을 것 같네요.
Issue 에 수정 사항 등록하고 다음 PR 에 진행 해 주세요

f-lab-bradley commented on src/main/java/com/inmybook/adapter/out/InMemoryPostRepository.java:
```diff
@@ -0,0 +1,21 @@
+
+@Repository
+public class InMemoryPostRepository implements RegisterPostPort {
+	private final HashMap<Long, Post> memory = new HashMap<>();
```
Get 할 때 ID 는 어떻게 찾나요???

f-lab-bradley reviewed (approved):
이 PR 은 여기서 일단 닫을게요.
PR 단위가 너무 커요.
개발 하나당 PR 하나로 하면 좋을 것 같아요.
